server/models/Notification.ts

```diff
@@ -218,6 +225,63 @@ class Notification extends Model<
   public get pixelUrl() {
     return `${env.URL}/api/notifications.pixel?token=${this.pixelToken}&id=${this.id}`;
   }
+
+  /**
+   * Returns the past message ids which are used to setup the thread chain in email clients.
+   *
+   * @param notification Notification for which the past notifications are fetched - used for determining the properties that form a thread.
+   * @returns An array of email message ids that form a thread.
+   */
+  public static async emailReferences(
+    notification: Notification
+  ): Promise<string[] | undefined> {
+    if (!isEmailThreadSupportedNotification(notification.event)) {
+      return;
+    }
+
+    const prevNotifications: Notification[] = [];
+    const limit = 100;
+    let offset = 0,
+      hasMore = true;
+
+    while (hasMore) {
+      const loadedNotifications = await this.unscoped().findAll({
+        attributes: ["id"],
+        where: {
+          id: {
+            [Op.ne]: notification.id,
+          },
+          event: notification.event,
+          documentId: notification.documentId,
```

Comment:
This isn't quite right as you'll start a separate thread for each notification type, we talked about having two separate threads – one for document edits and one for the conversation comment/mentions?

Reply:
This maintains separate threads for document edits, comments, and mentions.
Mention emails are in a separate thread since their `subject` is different from Comment emails. If we were to combine them, most email clients would dump the history in a single message instead of building the thread.

Essentially, Gmail and Outlook expect the `subject` to be consistent to present the thread tree in the UI (along with the necessary 'References' and `In-Reply-To` headers). So, I wouldn't suggest combining comment & mentions. Thoughts?

That said, this code has a bug - it doesn't combine the document mentions with the comment mentions (and creates 2 separate threads).

Reply:
> So, I wouldn't suggest combining comment & mentions. Thoughts?

Okay yes, it will be an improvement over today